core/src/kmx/kmx_plus.cpp

```diff
@@ -131,27 +131,40 @@ header_from_bytes(const uint8_t *data, KMX_DWORD length, uint32_t ident) {
  * @tparam T
  * @param data
  * @param length
+ * @param was_valid on exit, set to false if section was not valid
  * @return const T*
  */
 template <class T>
-const T *section_from_bytes(const uint8_t *data, KMX_DWORD length) {
+const T *section_from_bytes(const uint8_t *data, KMX_DWORD length, bool &was_valid) {
   if (length < sizeof(T)) { // Does not include dynamic data. First check.
     DebugLog("length < sizeof(section)");
     assert(false);
     return nullptr;
   }
   const COMP_KMXPLUS_HEADER *header = header_from_bytes(data, length, T::IDENT);
   const T *section = reinterpret_cast<const T *>(header);
-  if (section != nullptr && section->valid(length)) {
-    return section;
+  if (section != nullptr) {
+    if (!section->valid(length)) {
+      was_valid = false;
+      return nullptr;
+    } else {
+      // valid!
+      return section;
+    }
   } else {
+    // not found
     assert(false);
     return nullptr;
   }
```

Comment:
Can we refactor this into our normal pattern of early exit validation? (so that good path is not buried into an `else`).

Also could you add some comments as to the difference in handling between `section == nullptr` and `!section->valid(length)`?

```suggestion
  if (section == nullptr) {
    // not found
    assert(false);
    return nullptr;
  }
  
  if (!section->valid(length)) {
    was_valid = false;
    return nullptr;
  } 
  
  // valid!
  return section;
```